dask/dataframe/methods.py

```diff
@@ -78,7 +78,7 @@ def boundary_slice(
     Columns: []
     Index: []
     """
-    if df.empty:
+    if len(df.index) == 0:
```

Comment:
Are these not equivalent statements?

Reply:
```
pd.DataFrame({'a': [1,2,3]}).set_index('a').empty == True
pd.DataFrame({'a': [1,2,3], 'b':[1,2,3]}).set_index('a').empty == False
```

Reply:
Oh, that's not intuitive. Thanks for the info!

Reply:
yep, there is a mess with index-only/no-columns/no-rows/empty (and dtypes in these cases) in df